donnie_contrib/Events_Notify.py

```diff
@@ -0,0 +1,274 @@
+        required=True
+    )
+    description = discord.ui.TextInput(
+        label="Description",
+        placeholder="Enter a brief description of the event",
+        style=discord.TextStyle.long,
+        max_length=500,
+        required=False
+    )
+
+    async def on_submit(self, interaction: discord.Interaction):
+        await interaction.response.defer(thinking=True, ephemeral=True)
+
+        try:
+            # Parse and localize the user's input time (e.g., US/Eastern)
+            local_start_time = local_tz.localize(datetime.strptime(self.start_time.value, "%Y-%m-%d %H:%M"))
+            local_end_time = local_tz.localize(datetime.strptime(self.end_time.value, "%Y-%m-%d %H:%M"))
+
+            # Ensure the start time is in the future
+            if local_start_time < datetime.now(local_tz):
+                await interaction.followup.send("❌ The start time cannot be in the past.", ephemeral=True)
+                return
+
+            # Convert the start and end times to UTC for the Discord API
+            start_time_utc = local_start_time.astimezone(timezone('UTC')).isoformat()
+            end_time_utc = local_end_time.astimezone(timezone('UTC')).isoformat()
+
+        except ValueError:
+            await interaction.followup.send("❌ Invalid date format. Please use 'YYYY-MM-DD HH:MM'.", ephemeral=True)
+            return
+
+        discord_events = DiscordEvents()
+
+        # Send the event creation request to the API with UTC times
+        result = await discord_events.create_guild_event(
+            guild_id=interaction.guild_id,
+            event_name=self.event_name.value,
+            start_time=start_time_utc,  # Use UTC times
+            end_time=end_time_utc,  # Use UTC times
+            location=self.location.value,
+            description=self.description.value or "No description provided."
+        )
+
+        # Store the event information temporarily, we'll add the image URL later
+        event_data_store[interaction.id] = {
+            "event_name": self.event_name.value,
+            "start_time": start_time_utc,
+            "location": self.location.value,
+            "description": self.description.value or "No description provided.",
+            "image_url": None  # Will be updated later
+        }
+
+        await interaction.followup.send(result, ephemeral=True)
+
+        # Ask the user for an image URL after the event is created (ephemeral follow-up)
+        await interaction.followup.send("Event created! If you'd like to attach an image, please click the button below.", ephemeral=True, view=ImageInputView(interaction.id))
+
+
+# View for adding image button
+class ImageInputView(discord.ui.View):
+    def __init__(self, interaction_id):
+        super().__init__()
+        self.interaction_id = interaction_id
+
+    @discord.ui.button(label="Add Image", style=discord.ButtonStyle.primary)
+    async def add_image(self, interaction: discord.Interaction, button: discord.ui.Button):
```

Comment:
button arg doesnt appear to be used and may possibly be removed

Reply:
updating this to button to add a cover image instead of embedding an image into the notifications.